source/Octopus.Client/Repositories/Async/BuildInformationRepository.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Octopus.Client.Exceptions;
+using Octopus.Client.Logging;
+using Octopus.Client.Model;
+using Octopus.Client.Model.BuildInformation;
+
+namespace Octopus.Client.Repositories.Async
+{
+    class BuildInformationRepository : IBuildInformationRepository
+    {
+        private readonly IOctopusAsyncRepository repository;
+        private static readonly ILog Logger = LogProvider.For<BuildInformationRepository>();
+
+        public BuildInformationRepository(IOctopusAsyncRepository repository)
+        {
+            this.repository = repository;
+        }
+
+        public async Task<OctopusPackageVersionBuildInformationMappedResource> Get(string id)
+        {
+            var link = await repository.Link("BuildInformation");
+            return await repository.Client.Get<OctopusPackageVersionBuildInformationMappedResource>(link, new { id });
+        }
+
+        public Task<OctopusPackageVersionBuildInformationMappedResource> Push(string packageId, string version, OctopusBuildInformation octopusMetadata, bool replaceExisting)
+        {
+            return Push(packageId, version, octopusMetadata, replaceExisting ? OverwriteMode.OverwriteExisting : OverwriteMode.FailIfExists);
+        }
+        
+        public async Task<OctopusPackageVersionBuildInformationMappedResource> Push(string packageId, string version, OctopusBuildInformation octopusMetadata, OverwriteMode overwriteMode)
+        {
+            if (string.IsNullOrWhiteSpace(packageId))
+                throw new ArgumentException("A package Id must be supplied", nameof(packageId));
+            if (string.IsNullOrWhiteSpace(version))
+                throw new ArgumentException("A version must be supplied", nameof(version));
+
+            var resource = new OctopusPackageVersionBuildInformationResource
+            {
+                PackageId = packageId,
+                Version = version,
+                OctopusBuildInformation = octopusMetadata
+            };
+
+            if (!(await repository.HasLink("BuildInformation")))
+            {
+                throw new OperationNotSupportedByOctopusServerException(
+                    OctopusBuildInformation.BuildInformationRequiresOctopusVersionMessage,
+                    OctopusBuildInformation.BuildInformationRequiresOctopusVersion);
+            }
```

Comment:
Same here about creating the resource **only** if `BuildInformation` link exists?